Fix fixture table name derivation to respect custom Inflector rules

dereuromark opened the pull request:
## Summary
- Fixes #19124 - Custom Inflector rules (like uninflected) are not respected when fixtures derive their table names

The change in #18959 and follow-up #18972 changed fixture table name derivation from using `tableize()` (underscore then pluralize) to pluralizing the CamelCase class name directly. This broke custom Inflector uninflected rules since those rules match against underscored strings.

**Example:** With `Inflector::rules('uninflected', ['invoice_tax_information'])`, the fixture `InvoiceTaxInformationFixture` would incorrectly use `invoice_tax_informations` instead of `invoice_tax_information`.

**Old behavior (5.2.x):**
- `InvoiceTaxInformation` → `underscore()` → `invoice_tax_information` → `pluralize()` → `invoice_tax_information` (uninflected rule matches!)

**Broken behavior (5.3.0rc2):**
- `InvoiceTaxInformation` → `pluralize()` → `InvoiceTaxInformations` → `underscore()` → `invoice_tax_informations`

**Fix:**
Changes `_aliasFromClass()` to use `tableize()` then `camelize()`, which respects custom Inflector rules by doing underscore-then-pluralize (matching how uninflected rules are defined).

dereuromark committed:
Fix fixture table name derivation to respect custom Inflector rules

The change in #18959 and #18972 changed fixture table name derivation
from using tableize() (underscore then pluralize) to pluralizing the
CamelCase class name directly. This broke custom Inflector uninflected
rules since those rules match against underscored strings.

For example, with `Inflector::rules('uninflected', ['invoice_tax_information'])`,
the fixture `InvoiceTaxInformationFixture` would incorrectly use
`invoice_tax_informations` instead of `invoice_tax_information`.

This fix changes _aliasFromClass() to use tableize() then camelize(),
which respects custom Inflector rules by doing underscore then pluralize
(matching how uninflected rules are defined).

Fixes #19124

markstory commented on tests/TestCase/TestSuite/TestFixtureTest.php:
```diff
@@ -98,6 +100,33 @@ public function testAliasPlural(): void
+
+        // Add a custom uninflected rule
+        Inflector::rules('uninflected', ['invoice_tax_information']);
+        $this->assertSame('invoice_tax_information', Inflector::pluralize('invoice_tax_information'));
+        $this->assertSame('invoice_tax_information', Inflector::tableize('InvoiceTaxInformation'));
+
+        // The alias should be derived using tableize + camelize to respect uninflected rules
+        $this->assertSame(
+            'InvoiceTaxInformation',
+            Inflector::camelize(Inflector::tableize('InvoiceTaxInformation')),
+        );
+    }
```
This doesn't test anything with fixtures. Could we use anonymous classes to better simulate the regression?

dereuromark committed:
Improve test to use actual fixture behavior

Use EquipmentFixture to test uninflected word handling directly.
'equipment' is in the default uninflected list, so this tests
the actual fixture behavior rather than just the Inflector.
